bin/varnishd/cache/cache_req_fsm.c

```diff
@@ -1213,6 +1282,7 @@ CNT_Request(struct req *req)
 	}
 	wrk->vsl = NULL;
 	if (nxt == REQ_FSM_DONE) {
+		ReqFiniObjcoreStash(req);
 		INIT_OBJ(ctx, VRT_CTX_MAGIC);
 		VCL_Req2Ctx(ctx, req);
 		if (IS_TOPREQ(req)) {
```

Comment:
It's not safe to clear the stash before the privs free callbacks run.

Reply:
Hm. The free callbacks _should_ not do anything with references which the priv might still hold, but still I agree that changing the order is the safer option. Thank you for your valid point and good suggestion!

Reply:
What I have in mind is collecting something during the task, coming from a header for example, and logging it when the task ends. We replaced the `free()` callback with a `fini()` one that takes a `VRT_CTX` to allow privs to do things.